# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/svg/SVGIntegrationUtils.cpp
# Commit: 0fdbf0a455f2
# Full Hash: 0fdbf0a455f26afa0a7a0390d6a380e318fce9d4
# Author: Robert Longson <longsonr@gmail.com>
# Date: 2023-10-24 03:54:44
# Regressor Bug: 1765202
# File Overlap Count: 1
# Description:
#   Bug 1765202 part 3 - ignore clip-path and mask properties that don't point to clipPath and mask elements r=emilio
#   
#   This refactors the logic for testing whether something has a clip-path or mask and what to do about it in one place rather than using many overlapping functions.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D190767
# ==============================================================================

diff -r 2a8d11fcc746 -r 0fdbf0a455f2 layout/svg/SVGIntegrationUtils.cpp
--- a/layout/svg/SVGIntegrationUtils.cpp	Mon Oct 23 20:32:20 2023 +0000
+++ b/layout/svg/SVGIntegrationUtils.cpp	Mon Oct 23 20:32:37 2023 +0000
@@ -163,26 +163,6 @@
          style->HasClipPath() || style->HasMask();
 }
 
-bool SVGIntegrationUtils::UsingMaskOrClipPathForFrame(const nsIFrame* aFrame) {
-  const nsStyleSVGReset* style = aFrame->StyleSVGReset();
-  return style->HasClipPath() || style->HasMask();
-}
-
-bool SVGIntegrationUtils::UsingSimpleClipPathForFrame(const nsIFrame* aFrame) {
-  const nsStyleSVGReset* style = aFrame->StyleSVGReset();
-  if (!style->HasClipPath() || style->HasMask()) {
-    return false;
-  }
-
-  const auto& clipPath = style->mClipPath;
-  if (!clipPath.IsShape()) {
-    return false;
-  }
-
-  const auto& shape = clipPath.AsShape()._0;
-  return shape->IsRect() || shape->IsCircle() || shape->IsEllipse();
-}
-
 nsPoint SVGIntegrationUtils::GetOffsetToBoundingBox(nsIFrame* aFrame) {
   if (aFrame->HasAnyStateBits(NS_FRAME_SVG_LAYOUT)) {
     // Do NOT call GetAllInFlowRectsUnion for SVG - it will get the
@@ -742,9 +722,12 @@
 template <class T>
 void PaintMaskAndClipPathInternal(const PaintFramesParams& aParams,
                                   const T& aPaintChild) {
-  MOZ_ASSERT(SVGIntegrationUtils::UsingMaskOrClipPathForFrame(aParams.frame),
+#ifdef DEBUG
+  const nsStyleSVGReset* style = aParams.frame->StyleSVGReset();
+  MOZ_ASSERT(style->HasClipPath() || style->HasMask(),
              "Should not use this method when no mask or clipPath effect"
              "on this frame");
+#endif
 
   /* SVG defines the following rendering model:
    *